src/adapters/manga-stream.js

```diff
@@ -0,0 +1,167 @@
+  },
+
+  supportsReading() {
+    return true;
+  },
+
+  parseUrl(url) {
+    // https://readms.net/manga/attack_on_titan
+    // https://readms.net/r/attack_on_titan/103/4949/3
+
+    const u = utils.parseUrl(url);
+    const parts = u.pathname.split('/').filter(Boolean);
+    const isChapter = u.pathname.startsWith('/r/');
+
+    invariant(parts.length > 1, new errors.InvalidUrlError(url));
+
+    let seriesSlug = parts[1];
+    let chapterSlug = null;
+
+    if (isChapter) {
+      invariant(parts.length > 3, new errors.InvalidUrlError(url));
+      chapterSlug = getChapterSlugFromPathname(u.pathname);
+    }
+
+    return { seriesSlug, chapterSlug };
+  },
+
+  constructUrl(seriesSlug, chapterSlug) {
+    const initial = chapterSlug ? 'r' : 'manga';
+    const slug = [initial, seriesSlug, chapterSlug].filter(Boolean).join('/');
+
+    return utils.normalizeUrl(`https://readms.net/${slug}`);
+  },
+
+  async getSeries(seriesSlug) {
+    const url = this.constructUrl(seriesSlug);
+
+    const html = await throttledGet(url);
+    const dom = cheerio.load(html);
+
+    const title = dom('.main-body h1')
+      .text()
+      .trim();
+
+    // NOTE: MangaStream returns a 200 status for all pages, even when the
+    // slug doesn't exist. The "Page Not Found" h1 tag is our clue it's a 404.
+    invariant(title !== 'Page Not Found', new errors.NotFoundError(url));
+
+    const chapters = dom('.main-body h1 + table.table tr')
+      .slice(1) // skip the header row
+      .get()
+      .map(el => {
+        const row = dom(el);
+
+        const link = row.find('td > a');
+        const right = row.find('td:last-child');
+
+        const titleParts = link.text().split(' - ');
+        const chapterNumber = trimChapterNumber(titleParts[0]);
+        const title = titleParts[1];
+
+        const slug = getChapterSlugFromPathname(link.attr('href'));
+        const url = this.constructUrl(seriesSlug, slug);
+        const createdAt = getChapterTimestamp(right.text());
+
+        return { slug, title, url, createdAt, chapterNumber };
+      });
+
+    return { slug: seriesSlug, url, title, chapters };
+  },
+
+  async getChapter(seriesSlug, chapterSlug) {
+    const url = this.constructUrl(seriesSlug, chapterSlug);
+
+    const html = await utils.getPage(url);
+    const dom = cheerio.load(html);
+
+    const pageLinkList = dom('.btn-reader-page ul.dropdown-menu li a');
+    const firstPagePathname = pageLinkList.first().attr('href');
+    const lastPagePathname = pageLinkList.last().attr('href');
+
+    const firstPageNumber = getPageNumberFromPathname(firstPagePathname);
+    const lastPageNumber = getPageNumberFromPathname(lastPagePathname);
+
+    const getPageUrl = pageNumber =>
+      `${this.constructUrl(seriesSlug, chapterSlug)}/${pageNumber}`;
+    const pageUrls = utils
+      .range(firstPageNumber, lastPageNumber)
+      .map(getPageUrl);
+
+    const pages = await Promise.all(pageUrls.map(url => getPageFromURL(url)));
```

Comment:
This adapter requires loading all the pages, and then each image from that page. That said, it's surprisingly fast. I've included the image size probing code in `getPageFromURL` as a result.